src/integrationTest/java/com/mongodb/hibernate/junit/MongoExtension.java

```diff
@@ -70,18 +70,20 @@ public void preDestroyTestInstance(ExtensionContext context) {
 
     @Override
     public void beforeAll(ExtensionContext context) {
-        findAnnotatedFields(context.getRequiredTestClass(), InjectMongoCollection.class, field -> true)
-                .forEach(field -> {
-                    var annotation = field.getDeclaredAnnotation(InjectMongoCollection.class);
-                    String collectionName = annotation.value();
-                    var mongoCollection = mongoDatabase.getCollection(collectionName, BsonDocument.class);
-                    try {
-                        field.setAccessible(true);
-                        field.set(isStatic(field) ? null : context.getRequiredTestInstance(), mongoCollection);
-                    } catch (Exception ex) {
-                        throw new RuntimeException(ex);
-                    }
-                });
+        if (mongoDatabase != null) {
+            findAnnotatedFields(context.getRequiredTestClass(), InjectMongoCollection.class, field -> true)
+                    .forEach(field -> {
+                        var annotation = field.getDeclaredAnnotation(InjectMongoCollection.class);
+                        String collectionName = annotation.value();
+                        var mongoCollection = mongoDatabase.getCollection(collectionName, BsonDocument.class);
+                        try {
+                            field.setAccessible(true);
+                            field.set(isStatic(field) ? null : context.getRequiredTestInstance(), mongoCollection);
+                        } catch (Exception ex) {
+                            throw new RuntimeException(ex);
+                        }
+                    });
+        }
```

Comment:
Just found if our `MongoExtension` is used without PER_CLASS lifecycle, it would be broken. Currently just add a simple sanitfy check to avoid NPE, in the future we might introduce an annotation (analogous to `@SessionFactory` and `@Jpa` in `hibernate-testing`) in lieu of direct extension usage, so we could not only bind `@TestInstance( TestInstance.Lifecycle.PER_CLASS )` together, but also enjoy annotation elements and configurability.

Reply:
Just adding the `if (mongoDatabase != null)` does not make the current code correct when `Lifecycle.PER_CLASS` is used.

I propose https://github.com/stIncMale/mongo-hibernate/commit/10ebd61558d4455ac22b3c9c716846a26d41aff1, where I not only changed `MongoExtension`, but also made all other changes I am proposing, regardless of whether I made a comment about them or not).

Not only the proposed `MongoExtension` works regardless of the `Lifecycle` chosen, but it also uses the same `MongoClient` for all integration tests, which is something I wanted us to have from the get go, because this should save us noticeable time when running integration tests.

Reply:
Do you have objections to the change to `InjectMongoCollection.java`, or was it not applied accidentally?

Reply:
I overlooked. Added.